components/intake.py

```diff
@@ -31,61 +33,40 @@ def on_disable(self):
         """Run at the end of every control loop iteration."""
-        if self.motor_on == 1:
-            self.intake_left.set(1)
-        elif self.motor_on == -1:
-            self.intake_left.set(-1)
-        else:
-            self.intake_left.stopMotor()
-
-        if self.clamp_on:
-            self.clamp_arm.set(True)
-        else:
-            self.clamp_arm.set(False)
-
-        if self.push_on:
-            self.intake_kicker.set(True)
-        else:
-            self.intake_kicker.set(False)
-
-        if self.extension_on:
-            self.extension_arms.set(True)
-        else:
-            self.extension_arms.set(False)
-
-        self.seeing_cube()
-
-    def rotate(self, value):
-        """Turns the intake motors on or off."""
-        self.motor_on = value
-
-    def intake_clamp(self, value):
-        """Turns intake clamp on or off."""
+        self.left_motor.set(self.motor_output)
+        self.clamp_arm.set(self.clamp_on)
+        self.intake_kicker.set(self.push_on)
+        self.extension_arms.set(self.extension_on)
+
+        # Don't run the motors unless something else commands us to.
+        self.motor_output = 0
+        # We're not resetting the pneumatics outputs here, as once they
+        # have actuated, they stay actuated without drawing power.
+
+    def rotate(self, value: float):
+        """Set the output of the intake motors."""
+        self.motor_output = value
+
+    def clamp(self, value: bool):
+        """Turn the intake clamp on or off."""
         self.clamp_on = value
 
-    def intake_push(self, value):
-        """Turns the pushing pneumatic on or off."""
+    def push(self, value: bool):
+        """Turn the pushing pneumatic on or off."""
         self.push_on = value
 
-    def extension(self, value):
-        """Turns the extension pneumatics on or off."""
+    def extend(self, value: bool):
+        """Turn the extension pneumatics on or off."""
         self.extension_on = value
 
-    def infrared_distance(self):
-        """Gets the distance of the infrared sensor in m."""
-        self.cube_distance = self.infrared.getDistance() / 100
-        return self.cube_distance
+    def get_cube_distance(self) -> float:
+        """Get the distance of the infrared sensor in m."""
+        return self.infrared.getDistance() / 100
 
-    def seeing_cube(self):
-        """Returns True when the infrared sensor reads between 0.1m to 0.15m."""
-        if 0.1 <= self.cube_distance <= 0.15:
-            return True
-        else:
-            return False
+    def is_seeing_cube(self) -> bool:
+        """Check whether the infrared sensor reads between 0.1m to 0.15m."""
+        return 0.1 <= self.get_cube_distance() <= 0.15
 
-    def contacting_cube(self):
-        """Returns True of the current output of the motor is above 5."""
-        if self.intake_left.getOutputCurrent() >= 5:
-            return True
-        else:
-            return False
+    def is_contacting_cube(self) -> bool:
```

Comment:
What is contacting the cube? This name is still misleading as the intake is still contacting the cube after we suck it in, but this state will return false.

Reply:
Well, at least the docstring actually describes what this is actually used for now.

What would be a more appropriate and concise name for this method?

Reply:
Something like `wheels_contacting_cube` or something?

Reply:
It returns a boolean, so it should be phrased in the form of a question. `are_wheels_contacting_cube` seems unwieldy.

Reply:
What if we made some of these single line functions into properties?

Reply:
Not really much of a fan of using a property here. Not sure how expensive the getOutputCurrent call is.